internal/slicer/slicer.go

```diff
@@ -424,31 +417,31 @@ func selectPkgArchives(archives map[string]archive.Archive, selection *setup.Sel
 			continue
 		}
 		pkg := selection.Release.Packages[s.Package]
+
 		// If the package has not pinned any archive, choose the highest
 		// priority archive in which the package exists.
+		var candidates []*setup.Archive
 		if pkg.Archive == "" {
-			var chosen archive.Archive
-			for _, archiveInfo := range sortedArchives {
-				archive := archives[archiveInfo.Name]
-				if archive != nil && archive.Exists(pkg.Name) {
-					chosen = archive
-					break
-				}
-			}
-			if chosen == nil {
-				return nil, fmt.Errorf("slice package %q missing from archive(s)", pkg.Name)
-			}
-			pkgArchives[pkg.Name] = chosen
+			candidates = sortedArchives
 		} else {
-			archive := archives[pkg.Archive]
-			if archive == nil {
+			if archive, ok := archives[pkg.Archive]; !ok || archive == nil {
 				return nil, fmt.Errorf("archive %q not defined", pkg.Archive)
```

Comment:
I remember we talking about moving archive validation to when reading the release and not do it here, what is the status on that? See https://github.com/letFunny/chisel/pull/12#discussion_r1679487853.

Reply:
Sorry about that, changed in eef309d.